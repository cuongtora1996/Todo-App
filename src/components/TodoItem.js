import React, {Component} from 'react';
import {View, StyleSheet, Text, TouchableOpacity} from 'react-native';

class TodoItem extends Component {
  render() {
    const {text, isCompleted, onCheckComplete, onRemove} = this.props.item;
    return (
      <View style={styles.container}>
        <TouchableOpacity
          onPress={() => onCheckComplete && onCheckComplete(this.props.item)}
          style={styles.checkboxWrapper}>
          <View style={styles.checkbox}>
            <View
              style={[
                styles.check,
                {
                  backgroundColor: isCompleted ? '#000' : '#FFF',
                },
              ]}
            />
          </View>
        </TouchableOpacity>
        <Text style={styles.text}>{text}</Text>
        <TouchableOpacity
          onPress={() => onRemove && onRemove(this.props.item.id)}
          style={styles.deleteWrapper}>
          <View style={styles.deleteButton}>
            <Text style={styles.deleteText}>Delete</Text>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginHorizontal: 12,
    paddingVertical: 12,
    borderBottomColor: "#AEAEAE",
    borderBottomWidth: 0.5,
  },
  checkboxWrapper: {
    flexShrink: 1,
    marginRight: 12,
  },
  checkbox: {
    height: 24,
    width: 24,
    borderWidth: 2,
    borderColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
  },
  check: {
    height: 12,
    width: 12,
  },
  text: {
    flexGrow: 1,
    fontSize: 16,
    color: '#000',
    fontWeight: 'bold',
  },
  deleteWrapper: {
    flexShrink: 1,
    alignSelf: 'flex-end',
  },
  deleteButton: {
    backgroundColor: '#CC3300',
    padding: 8,
    borderRadius: 8,
  },
  deleteText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  }
});

export default TodoItem;
