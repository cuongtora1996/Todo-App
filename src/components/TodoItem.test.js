import React from 'react';
import TodoItem from './TodoItem';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer
    .create(<TodoItem item={{text: "Item 1", isCompleted:false }}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
