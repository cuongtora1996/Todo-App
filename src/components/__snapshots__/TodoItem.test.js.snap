// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#AEAEAE",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 12,
      "paddingVertical": 12,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexShrink": 1,
        "marginRight": 12,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#000",
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 12,
              "width": 12,
            },
            Object {
              "backgroundColor": "#FFF",
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#000",
        "flexGrow": 1,
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Item 1
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#CC3300",
          "borderRadius": 8,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Delete
      </Text>
    </View>
  </View>
</View>
`;
